### PR TITLE
Add read only to noop functions

### PR DESCRIPTION
**Pull request by morgo**
### What is changed, added or deleted? (Required)

Add documentation for https://github.com/pingcap/tidb/pull/23818

This affects master only.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -451,6 +451,8 @@ Constraint checking is always performed in place for pessimistic transactions (d
     * `SQL_CALC_FOUND_ROWS` syntax
     * `CREATE TEMPORARY TABLE` syntax
     * `DROP TEMPORARY TABLE` syntax
+    * `START TRANSACTION READ ONLY` and `SET TRANSACTION READ ONLY` syntax
+    * The `tx_read_only`, `transaction_read_only`, `offline_mode`, `super_read_only`and `read_only` system variables
```
Comment (TomShawn) [inline system-variables.md:455]:
```suggestion
    * The `tx_read_only`, `transaction_read_only`, `offline_mode`, `super_read_only` and `read_only` system variables
```

**Review by TomShawn — COMMENTED** [review]
Which release version(s) will this content be cherry-picked to?

**Comment by morgo** [comment]
> Which release version(s) will this content be cherry-picked to?

This is for master only.

**Comment by TomShawn** [comment]
> > Which release version(s) will this content be cherry-picked to?
> 
> This is for master only.

I guess the content for master will be applicable to release versions in the future. If we do not update the release version once it is applicable, the content is simply missing in that branch. Since this PR has the `for-future-release` label, please remember to make the cherry-pick once it is ready. Thanks!

**Pull request merged**